evalml/pipelines/pipeline_base.py

```diff
@@ -439,31 +441,44 @@ def graph_json(self):
         Returns:
             dag_json (str): A serialized JSON representation of a DAG structure.
         """
-        nodes = {
-            comp_: {"Attributes": att_.parameters, "Name": att_.name}
-            for comp_, att_ in self.component_graph.component_instances.items()
-        }
-
-        x_edges = self.component_graph._get_edges(
+        nodes = {}
+        for comp_, att_ in self.component_graph.component_instances.items():
+            param_dict = {}
+            for param, val in att_.parameters.items():
+                # Can't JSON serialize components directly, have to split them into name and parameters
+                if isinstance(val, ComponentBase):
+                    param_dict[f"{param}_name"] = val.name
+                    param_dict[f"{param}_parameters"] = val.parameters
+                else:
+                    if isinstance(val, np.int64):
+                        val = int(val)
```

Comment:
This is the fix for the `int64` bug. There is still room for a problem to occur here, if there is a tuned integer hyperparameter within an ensembling component. However, the default final estimator at this point is `ElasticNet`, which doesn't have any integer hyperparameters, so this won't be an issue in the default case.

I think we should leave this for another issue, where we move all the `Integer` hyperparameter ranges to return `int`s instead of `np.int64`, but if people prefer I could add another catch in here for ensemble parameters.

Reply:
Thanks for catching this, I say go for it. If users want to change the final estimator in ensembling this should at least give them an idea of why they can't get the `graph` for their component graph just yet.